core/tests/test_utils.py

```diff
@@ -717,22 +719,19 @@ def save_new_linear_exp_with_state_names_and_interactions(
         interaction_ids = itertools.cycle(interaction_ids)
 
         exploration = exp_domain.Exploration.create_default_exploration(
-            exploration_id, title=title, category=category,
-            language_code=language_code)
-        exploration.objective = objective
-        exploration.add_states(state_names)
-
-        from_names = [exploration.init_state_name] + state_names[:-1]
-        to_names = state_names
-        for from_name, to_name in zip(from_names, to_names):
-            from_state = exploration.states[from_name]
-            from_state.update_interaction_id(next(interaction_ids))
-            from_state.interaction.default_outcome.dest = to_name
-
-        # Prepare end_state.
-        end_state = exploration.states[to_names[-1]]
-        end_state.update_interaction_id('EndExploration')
-        end_state.interaction.default_outcome = None
+            exploration_id, title=title, init_state_name=state_names[0],
+            category=category, objective=objective, language_code=language_code)
+
+        if len(state_names) > 1:
```

Comment:
This doesn't seem to handle the 1-state case correctly, right? Don't you need to have an EndExploration set up for that?

Reply:
I took a look at  the `create_default_exploration` code, and it seems like it handles the end state properly for you. I was thinking maybe this function only makes sense to use over the other with >2 states, but I didn't really see the need to be so restricting.

Reply:
Sorry, not sure I understand. In the previous version of this code, the end state was prepared (lines 732-735) regardless of whether the exploration had 1 or more states. Here, you only prepare it if len(state_names) > 1. I was wondering if we should move that part out of the if condition, because otherwise you're going to end up with an exploration with one state whose interaction is None, and that's not valid/publishable.

Reply:
Gotcha, so when `create_default_exploration` is called, it actually creates 2 states for us: the init state and the end state. That's why I choose not to do any set-up in the case exactly 1 name is given. I guess I could make the change, but does it make sense to have an exploration that ends immediately?

Reply:
1-state explorations don't make sense pedagogically, but they are a valid construct.

I just looked at create_default_exploration again. Can you explain why you think it creates two states? I only see one in the states dict.

Reply:
Ahhh, I see my mistake. I thought `default_dest_state_name` was used to create a second state, but I see now that it's actually used to create a "loop" by default. OK, I'll add a fix!

Reply:
Done.